plugins/inputs/socket_listener/socket_listener.go

```diff
@@ -47,6 +48,27 @@ func (ssl *streamSocketListener) listen() {
 		}
 		ssl.connections[c.RemoteAddr().String()] = c
 		ssl.connectionsMtx.Unlock()
+
+		if ssl.KeepAlivePeriod != nil {
+			if tcpc, ok := c.(*net.TCPConn); ok {
+				if ssl.KeepAlivePeriod.Duration == 0 {
+					if err := tcpc.SetKeepAlive(false); err != nil {
+						ssl.AddError(fmt.Errorf("unable to disable keep alive: %s", err))
+					}
+				} else {
+					if err := tcpc.SetKeepAlive(true); err != nil {
+						ssl.AddError(fmt.Errorf("unable to enable keep alive: %s", err))
+					} else {
+						if err := tcpc.SetKeepAlivePeriod(ssl.KeepAlivePeriod.Duration); err != nil {
+							ssl.AddError(fmt.Errorf("unable to set keep alive period: %s", err))
```

Comment:
Yeah, that was along the lines of the idea, just thought it might be a little odd breaking it out. But I went and implemented it (and pushed to PR) and I think it's clean.